availability/watcher.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2016: Mirantis Inc.
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import datetime as dt
+import json
+import logging
+import sys
+import threading
+import time
+
+import queue
+import requests
+
+from availability import config
+
+
+SERVICES = config.get_config().get("services")
+ELASTIC = config.get_config().get("elastic", "http://127.0.0.1:9200")
+SERVICE_CONN_TIMEOUT = config.get_config().get("connection_timeout", 1)
+SERVICE_READ_TIMEOUT = config.get_config().get("read_timeout", 10)
+
+LOG_LEVEL = config.get_config().get("logging", {}).get("level", "INFO")
+LOG = logging.getLogger("watcher")
+LOG.setLevel(LOG_LEVEL)
+
+
+def check_availability(data, results_queue):
+    """Check if service is available and put result to Queue.
+
+    :param data: dict that represents service query data
+    :param results_queue: Queue for results
+    """
+    try:
+        requests.get(data["url"], timeout=(SERVICE_CONN_TIMEOUT,
+                                           SERVICE_READ_TIMEOUT))
+        data["status"] = True
+    except requests.exceptions.RequestException:
+        data["status"] = False
+
+    results_queue.put(data)
+
+
+def save_availability(ts, data):
+    """Send availability data to Elastic.
+
+    :param ts: float timestamp
+    :param data: dict data to send
+    :rtype: bool
+    """
+    if not ELASTIC:
+        LOG.error("Elastic is not configured")
+        return
+    url = "%s/availability/ts/%f" % (ELASTIC, ts)
+    LOG.info("Saving: %s %r" % (url, data))
+    res = requests.put(url, data=json.dumps(data))
+    if res.status_code in (200, 201):
+        return True
+    LOG.error("Failed to save availability to Elastic (code %i): %s"
+              % (res.status_code, res.text))
+    return False
+
+
+def watch_services(period=None):
+    """Constantly check services availability.
+
+    This runs infinite services availability check with given period.
+    """
+    if not SERVICES:
+        LOG.error("No services configured. Quitting.")
+        return 1
+
+    period = period or config.get_config().get("period", 60)
+    LOG.info("Start watching with period %s seconds" % period)
+
+    if SERVICE_CONN_TIMEOUT + SERVICE_READ_TIMEOUT > period:
+        LOG.error("Period can not be lesser than timeout, "
+                  "otherwice threads could crowd round.")
+        return 1
+
+    results_queue = queue.Queue()
+
+    while True:
```

Comment:
let's use schedule library, I found it very nice: 
https://github.com/seecloud/health/blob/master/health/job.py#L158